vulnerabilities/importers/nvd.py

```diff
@@ -20,150 +20,175 @@
-            severity_scores.append(
-                VulnerabilitySeverity(
-                    system=scoring_systems["cvssv2"],
-                    value=str(cve_item["impact"]["baseMetricV2"]["cvssV2"]["baseScore"]),
-                )
-            )
-            severity_scores.append(
-                VulnerabilitySeverity(
-                    system=scoring_systems["cvssv2_vector"],
-                    value=str(cve_item["impact"]["baseMetricV2"]["cvssV2"]["vectorString"]),
-                )
+        references.append(
+            Reference(
+                url=f"https://nvd.nist.gov/vuln/detail/{cve_id}",
+                reference_id=cve_id,
+                severities=severity_scores,
             )
-
-        return severity_scores
-
-    def extract_reference_urls(self, cve_item):
-        urls = set()
-        for reference in cve_item["cve"]["references"]["reference_data"]:
-            ref_url = reference["url"]
-
-            if not ref_url:
-                continue
-
-            if ref_url.startswith("http") or ref_url.startswith("ftp"):
-                urls.add(ref_url)
-
-        return urls
-
-    def is_outdated(self, cve_item):
-        cve_last_modified_date = cve_item["lastModifiedDate"]
-        cve_last_modified_date_obj = dateparser.parse(cve_last_modified_date)
-
-        if self.config.cutoff_date:
-            return cve_last_modified_date_obj < self.config.cutoff_date
-
-        if self.config.last_run_date:
-            return cve_last_modified_date_obj < self.config.last_run_date
-
-        return False
-
-    def related_to_hardware(self, cve_item):
-        for cpe in self.extract_cpes(cve_item):
-            cpe_comps = cpe.split(":")
-            # CPE follow the format cpe:cpe_version:product_type:vendor:product
-            if cpe_comps[2] == "h":
-                return True
-
-        return False
-
-    @staticmethod
-    def extract_cpes(cve_item):
-        cpes = set()
-        for node in cve_item["configurations"]["nodes"]:
-            for cpe_data in node.get("cpe_match", []):
+        )
+        if "https://nvd.nist.gov/vuln/detail/{cve_id}" in ref_urls:
+            ref_urls.remove(f"https://nvd.nist.gov/vuln/detail/{cve_id}")
+        references.extend([Reference(url=url) for url in ref_urls])
+        if cve_id:
+            aliases.append(cve_id)
+        summary = extract_summary(cve_item)
+        yield AdvisoryData(
+            aliases=aliases,
+            summary=summary,
+            references=sorted(references),
+            date_published=dateparser.parse(cve_item.get("publishedDate")),
+        )
+
+
+def extract_reference_urls(cve_item):
+    """
+    Return a list of reference URLs for a given CVE item.
+    """
+    urls = set()
+    for reference in get_item(cve_item, "cve", "references", "reference_data") or []:
+        ref_url = reference.get("url")
+
+        if not ref_url:
+            continue
+
+        if ref_url.startswith("http") or ref_url.startswith("ftp"):
```

Comment:
Using a tuple is better IMHO
```suggestion
        if ref_url.startswith(("http", "ftp",)):
```